apps/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -688,4 +680,32 @@ export abstract class BaseInstallManager {
       }
     }
   }
+
+  protected async validateNpmSetup(): Promise<void> {
+    if (this._npmSetupValidated) {
+      return;
+    }
+
+    if (!this.options.bypassPolicy) {
+      const setupPackageRegistry: SetupPackageRegistry = new SetupPackageRegistry({
+        rushConfiguration: this.rushConfiguration,
+        isDebug: this.options.debug,
+        syncNpmrcAlreadyCalled: this._syncNpmrcAlreadyCalled
+      });
+      const valid: boolean = await setupPackageRegistry.checkOnly();
+      if (!valid) {
+        console.error();
+        console.error(colors.red('ERROR: NPM credentials are missing or expired'));
+        console.error();
+        console.error(
+          colors.bold(
+            '==> Please run "rush setup" to update your NPM token.  (Or append "--bypass-policy" to proceed anyway.)'
```

Comment:
Maybe we should make `--bypass-setup` rather than reusing `--bypass-policy`

Reply:
I've decided to tackle this in a separate PR.  We should separate it into two flags, one for bypassing "policies about how we work" (e.g. Git email) and another for bypassing "safeguards that prevent problems" (e.g. missing NPM token).